api/routers/report.py

```diff
@@ -1,65 +1,74 @@
 from typing import List, Optional
 
-from api.database.functions import (EngineType, get_session,
-                                    verify_token, sqlalchemy_result)
-from api.database.models import Player, Prediction, Report, ReportLatest
-from fastapi import APIRouter, HTTPException, Query
+import pandas as pd
+from api.Config import (back_time_buffer, front_time_buffer, report_maximum,
+                        upper_gear_cost)
+from api.database.functions import (EngineType, batch_function, create_task,
+                                    get_session, is_valid_rsn, list_to_string,
+                                    parse_detection, sql_insert_player,
+                                    sql_insert_report, sql_select_players,
+                                    sqlalchemy_result, to_jagex_name,
+                                    verify_token)
+from api.database.models import (Player, Prediction, Report, ReportLatest,
+                                 stgReport)
+from fastapi import APIRouter, HTTPException, Query, status, Request
+from fastapi.encoders import jsonable_encoder
+from fastapi.responses import JSONResponse
 from pydantic import BaseModel
 from sqlalchemy import update
-from sqlalchemy.sql.expression import insert, select
 from sqlalchemy.sql import func
-
+from sqlalchemy.sql.expression import insert, select
 
 logger = logging.getLogger(__name__)
 router = APIRouter()
 
-
 class equipment(BaseModel):
-    equip_head_id: int
-    equip_amulet_id: int
-    equip_torso_id: int
-    equip_legs_id: int
-    equip_boots_id: int
-    equip_cape_id: int
-    equip_hands_id: int
-    equip_weapon_id: int
-    equip_shield_id: int
-
-
-class detection(BaseModel):
-    reportedID: int
-    reportingID: int
-    region_id: int
-    x_coord: int
-    y_coord: int
-    z_coord: int
-    ts: int
-    manual_detect: int
-    on_members_world: int
-    on_pvp_world: int
-    world_number: int
+    equip_head_id: int = Query(None, ge=0)
+    equip_amulet_id: int = Query(None, ge=0)
+    equip_torso_id: int = Query(None, ge=0)
+    equip_legs_id: int = Query(None, ge=0)
+    equip_boots_id: int = Query(None, ge=0)
+    equip_cape_id: int = Query(None, ge=0)
+    equip_hands_id: int = Query(None, ge=0)
+    equip_weapon_id: int = Query(None, ge=0)
+    equip_shield_id: int = Query(None, ge=0)
+
+
+class detections(BaseModel):
+    reporter: str = Query(..., min_length=1, max_length=12)
+    reported: str = Query(..., min_length=1, max_length=12)
+    region_id: int = Query(0, ge=0, le=100_000)
+    x_coord: int = Query(0, ge=0)
+    y_coord: int = Query(0, ge=0)
+    z_coord: int = Query(0, ge=0)
+    ts: int = Query(0, ge=0)
+    manual_detect: int = Query(0, ge=0, le=1)
+    on_members_world: int = Query(0, ge=0, le=1)
+    on_pvp_world: int = Query(0, ge=0, le=1)
+    world_number: int = Query(0, ge=300, le=1_000)
     equipment: equipment
-    equip_ge_value: int
-
+    equip_ge_value: int = Query(0, ge=0, le=int(upper_gear_cost))
```

Comment:
don't think this is good to define

Reply:
Set the value very, very high already, so it will should not be a problem. No player feasibly has 1T gold to wear and the highest in our db is 9B.